From 74a33553ca1a684cb89d2ded049c16436b74c5bb Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Tue, 8 Apr 2025 21:26:00 +0800
Subject: [PATCH] workaround libatomic linkage for static libpython on riscv

---
 nuitka/build/Backend.scons | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/nuitka/build/Backend.scons b/nuitka/build/Backend.scons
index 112ca8cbbf98..f62bbf108686 100644
--- a/nuitka/build/Backend.scons
+++ b/nuitka/build/Backend.scons
@@ -14,6 +14,7 @@ build process for itself, although it can be compiled using the same method.
 
 import sys
 import os
+import platform
 import types
 
 sys.modules["nuitka"] = types.ModuleType("nuitka")
@@ -273,6 +274,9 @@ elif env.exe_mode or env.dll_mode:
     if env.python_prefix_external != "/usr" and isLinux():
         env.Append(LIBS=["dl", "pthread", "util", "rt", "m"])
 
+        if platform.machine().startswith("riscv"):
+            env.Append(LIBS=["atomic"])
+
         if env.gcc_mode:
             if env.clang_mode:
                 env.Append(LINKFLAGS=["-Wl,--export-dynamic"])
-- 
2.54.0

