From 4b06393e6e5fa5272550f0f646c7b0f66fa521da Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Tue, 8 Apr 2025 21:26:00 +0800
Subject: [PATCH] workaround libatomic linkage for static libpython on riscv

---
 nuitka/build/Backend.scons | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/nuitka/build/Backend.scons b/nuitka/build/Backend.scons
index a724006f70a6..889eaa49ae4e 100644
--- a/nuitka/build/Backend.scons
+++ b/nuitka/build/Backend.scons
@@ -13,6 +13,7 @@ build process for itself, although it can be compiled using the same method.
 # Make nuitka package importable from calling installation
 
 import os
+import platform
 import sys
 import types
 
@@ -275,6 +276,9 @@ elif env.exe_mode or env.dll_mode:
     if env.python_prefix_external != "/usr" and isLinux():
         env.Append(LIBS=["dl", "pthread", "util", "rt", "m"])
 
+        if platform.machine().startswith("riscv"):
+            env.Append(LIBS=["atomic"])
+
         if env.gcc_mode:
             if env.clang_mode:
                 env.Append(LINKFLAGS=["-Wl,--export-dynamic"])
-- 
2.55.0

