From 4e68c212461b228481be51ad52534369cfb623d7 Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Tue, 8 Apr 2025 21:26:00 +0800
Subject: [PATCH] workaround libatomic linkage for static libpython on riscv

---
 nuitka/build/Backend.scons | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/nuitka/build/Backend.scons b/nuitka/build/Backend.scons
index 6eb130b1b..c20ce8a47 100644
--- a/nuitka/build/Backend.scons
+++ b/nuitka/build/Backend.scons
@@ -14,6 +14,7 @@ build process for itself, although it can be compiled using the same method.
 
 import sys
 import os
+import platform
 import types
 
 sys.modules["nuitka"] = types.ModuleType("nuitka")
@@ -797,6 +798,9 @@ elif not env.module_mode:
     if python_prefix_external != "/usr" and "linux" in sys.platform:
         env.Append(LIBS=["dl", "pthread", "util", "rt", "m"])
 
+        if platform.machine().startswith("riscv"):
+            env.Append(LIBS=["atomic"])
+
         if env.gcc_mode:
             if clang_mode:
                 env.Append(LINKFLAGS=["-Wl,--export-dynamic"])
-- 
2.48.1

