From 3fdc249394b912d347165af9d4d1f9910f06c482 Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Tue, 8 Apr 2025 21:26:00 +0800
Subject: [PATCH 1/2] workaround libatomic linkage for static libpython on
 riscv

---
 nuitka/build/Backend.scons | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/nuitka/build/Backend.scons b/nuitka/build/Backend.scons
index f2f0016ba..ffc71984f 100644
--- a/nuitka/build/Backend.scons
+++ b/nuitka/build/Backend.scons
@@ -14,6 +14,7 @@ build process for itself, although it can be compiled using the same method.
 
 import sys
 import os
+import platform
 import types
 
 sys.modules["nuitka"] = types.ModuleType("nuitka")
@@ -760,6 +761,9 @@ elif env.exe_mode or env.dll_mode:
     if python_prefix_external != "/usr" and "linux" in sys.platform:
         env.Append(LIBS=["dl", "pthread", "util", "rt", "m"])
 
+        if platform.machine().startswith("riscv"):
+            env.Append(LIBS=["atomic"])
+
         if env.gcc_mode:
             if clang_mode:
                 env.Append(LINKFLAGS=["-Wl,--export-dynamic"])
-- 
2.48.1

